src/core/load-fragment.c

```diff
@@ -1364,12 +1365,22 @@ int config_parse_numa_mask(const char *unit,
                            void *userdata) {
         int r;
         NUMAPolicy *p = data;
+        _cleanup_free_ char *all = NULL;
 
         assert(filename);
         assert(lvalue);
         assert(rvalue);
         assert(data);
 
+        if (streq(rvalue, "all")) {
+                r = numa_mask_all_string(&all);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to determine maximum NUMA node index, falling back to 1024: %m");
+                        rvalue = "0-1024";
+                } else
+                        rvalue = all;
+        }
```

Comment:
hmm, it's a bit strange to first format a string here, which you then parse again. why not just set the bits directly?

Reply:
Making a string representation on "all NUMA nodes" and then treating it as any other NUMA node mask seemed better. Also setting bits in CPU mask needs to be done carefully and code is kinda ugly (have look in `parse_cpu_set_full()` so I didn't want to pollute the code for little gain).

As always, if you feel strongly about this I can rework. Let me know.

Reply:
it's pretty short, no, just do:

```c
int numa_mask_add_all(CPUSet *c) {
        int m, r;

        assert(c);

        m = numa_max_node();
        if (m < 0) {
                log_debug_errno(r, "Failed to determine maximum NUMA node index, assuming 8K: %m");
                m = 8191; /* At least on PPC this is the max number of nodes possible, on kenrel 5.1 */
        }

        r = cpu_set_realloc(c, m);
        if (r < 0)
                return r;

        for (int i = 0; i < m; i++) 
                CPU_SET_S(i, cpu_set->allocated, cpu_set->set);

        return 0;
}
```

That's pretty short and easy to read, no?

(the 8191 comes from the hardcoded limit in cpu_set_add(). maybe something to make a `#define` from)

Reply:
Yeah, that seems better (and is approximately the same amount of code).